enterprise_subsidy/apps/transaction/management/commands/write_reversals_from_enterprise_unenrollments.py

```diff
@@ -0,0 +1,209 @@
+        """
+        DRY_RUN_LOG_PREFIX = ""
+        if dry_run := options.get('dry_run'):
+            DRY_RUN_LOG_PREFIX = "DRY RUN: "
+            logger.info("Running in dry-run mode. No updates or creates will occur.")
+
+        logger.info(
+            f"{DRY_RUN_LOG_PREFIX}Updating and writing Transaction Reversals from recent Enterprise unenrollments data"
+        )
+        recent_unenrollments = EnterpriseApiClient().fetch_recent_unenrollments()
+        logger.info(
+            f"{DRY_RUN_LOG_PREFIX}Found {len(recent_unenrollments)} recent Enterprise unenrollments"
+        )
+
+        reversals_processed = 0
+        enterprise_catalog_api_client = EnterpriseCatalogApiClient()
+        fetched_content_metadata = {}
+        for unenrollment in recent_unenrollments:
+            fulfillment_uuid = unenrollment.get("uuid")
+            enterprise_course_enrollment = unenrollment.get("enterprise_course_enrollment")
+            enrollment_course_run_key = enterprise_course_enrollment.get("course_id")
+            enrollment_unenrolled_at = enterprise_course_enrollment.get("unenrolled_at")
+            try:
+                # Look for a transaction related to the unenrollment
+                # This will throw a Transaction.DoesNotExist exception if no transaction is found
+                related_transaction = Transaction.objects.get(uuid=unenrollment.get('transaction_id'))
+                idempotency_key = generate_transaction_reversal_idempotency_key(
+                    fulfillment_uuid,
+                    enrollment_unenrolled_at
+                )
+                # This ORM query will throw a Reversal.DoesNotExist exception if no reversal is found,
+                # which means we can move forward with the reversal generation
+                existing_reversal = Reversal.objects.get(
+                    transaction=related_transaction,
+                    idempotency_key=idempotency_key,
+                )
+            except Transaction.DoesNotExist:
+                logger.info(
+                    f"{DRY_RUN_LOG_PREFIX}No Subsidy Transaction found for enterprise fulfillment: {fulfillment_uuid}"
+                )
+                continue
+            except Reversal.DoesNotExist:
+                logger.info(
+                    f"{DRY_RUN_LOG_PREFIX}No existing Reversal found for enterprise fulfillment: {fulfillment_uuid}. "
+                    f"Writing Reversal for Transaction: {related_transaction}."
+                )
+
+                # On initial release we are only supporting learner initiated unenrollments for OCM courses.
+                # OCM courses are identified by the lack of an external_reference on the Transaction object.
+                # Externally referenced transactions can be unenrolled through the Django admin actions related to the
+                # Transaction model.
+                if not related_transaction.external_reference.exists():
```

Comment:
here's another place where an early continue will help reduce this nesting, e.g.
```python
if related_transaction.external_reference.exists():
    # do your logging
    continue
```